### PR TITLE
Package Source Mapping matches found, show no logs for projects

### PR DESCRIPTION
**Pull request by martinrrm**
## Bug

Fixes:
https://github.com/NuGet/Home/issues/11413
Regression? Last working version:

## Description

No logs should be displayed for Package Source Mapping matches for projects, only for packages.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [ ] N/A

### CODE REVIEW

**Review by jeffkl — APPROVED** [review]

File: src/NuGet.Core/NuGet.DependencyResolver.Core/ResolverUtility.cs [hunk 1]
```diff
@@ -47,7 +47,11 @@ public static class ResolverUtility
             var currentCacheContext = context.CacheContext;
 
             IList<IRemoteDependencyProvider> remoteDependencyProviders = context.FilterDependencyProvidersForLibrary(libraryRange);
-            LogIfPackageSourceMappingIsEnabled(libraryRange.Name, context, remoteDependencyProviders);
+
+            if (libraryRange.TypeConstraintAllows(LibraryDependencyTarget.Package))
+            {
+                LogIfPackageSourceMappingIsEnabled(libraryRange.Name, context, remoteDependencyProviders);
```
Comment (erdembayar) [inline src/NuGet.Core/NuGet.DependencyResolver.Core/ResolverUtility.cs:53]: There's [another ref](https://github.com/NuGet/NuGet.Client/blob/9187a50daec29061772a2057490697c6c0e4c354/src/NuGet.Core/NuGet.DependencyResolver.Core/ResolverUtility.cs#L240) of this method. Did you check it?

Comment (martinrrm) [reply src/NuGet.Core/NuGet.DependencyResolver.Core/ResolverUtility.cs:53]: Thanks! I missed this

Comment (erdembayar) [reply src/NuGet.Core/NuGet.DependencyResolver.Core/ResolverUtility.cs:53]: Does this issue repro on `packages.config` projects too? If yes, then I can point to where to fix them too.

Comment (martinrrm) [reply src/NuGet.Core/NuGet.DependencyResolver.Core/ResolverUtility.cs:53]: Not sure, can you show me? I can take a look and debug it to see if it is needed

Comment (erdembayar) [reply src/NuGet.Core/NuGet.DependencyResolver.Core/ResolverUtility.cs:53]: Just create `packages.config` projects and use same nuget.config file.

Comment (nkolev92) [reply src/NuGet.Core/NuGet.DependencyResolver.Core/ResolverUtility.cs:53]: It won't repro in packages.config. 

Packages.config doesn't use the projects at all.

**Review by nkolev92 — APPROVED** [review]

**Review by erdembayar — APPROVED** [review]

**Pull request merged**